app/views/moderations/mod.html.erb

```diff
@@ -155,6 +156,46 @@
       <% end %>
     </div>
   <% end %>
+
+  <% if @moderatable.class.name == "Article" && (current_user.has_role?(:super_admin) || current_user.has_role?(:tag_moderator, :any)) %>
+    <div class="tag-mod-form">
+      <h2> Add Appropriate Tags</h2>
+      <% if @moderatable.tag_list.count < 4 %>
+        <%= form_for(TagAdjustment.new) do |f| %>
+          <% unless current_user.has_role?(:super_admin) %>
+            <ul>
+              <li>You can only add tags that you moderate.</li>
+              <li>Only add tags that are appropriate.</li>
+              <li>Always be friendly in your tag adding reason.</li>
+            </ul>
+          <% end %>
+          <b>Current live tags:</b> <%= @moderatable.tag_list %>
+          <br /><br />
+          <%= f.hidden_field :article_id, value: @moderatable.id %>
+          <%= f.hidden_field :adjustment_type, value: "addition" %>
+          <%= f.text_field :tag_name, placeholder: "Tag Name - only add one at a time" %>
+          <%= f.text_area :reason_for_adjustment, placeholder: "Reason for addition (Be super kind) - Only the reason is needed, the notification will take care of the rest." %>
+          <%= f.submit "Add Tag" %>
```

Comment:
probably we should have some JS that enables this only if the user has provided both the tag and the reason

Reply:
I just added required: true to the tag_name and reason_for_adjustment to make sure they are filled out.. This will allow empty strings to get through but there are validations on the model. 
I could prevent this on the front-end with javascript but there are other object errors that are not being prevented or displayed. Right now if a mod enters a tag to remove that they are not moderating or that doesn't exist, it will cause a Validation failed: error.  I could refactor to catch and display these errors to the new tag adjustment form on the mod view.

Let me know if you want me to refactor or leave as is?

Reply:
I think it's okay! thank you!